### ISSUE TITLE
Support Firefox mobile

### ISSUE THREAD

**Issue by guerda** [stefanw/bibbot#393]
Since December 14, Firefox supports almost all add-ons from the desktop also on Mobile. [1]

It would be great if BibBot would also be available on Firefox mobile, which is described here: [2]

[1]: https://blog.mozilla.org/addons/2023/11/28/open-extensions-on-firefox-for-android-debut-december-14-but-you-can-get-a-sneak-peek-today/

[2]: https://extensionworkshop.com/documentation/develop/developing-extensions-for-firefox-for-android/

**Comment by Th-o-mas** [comment]
I am delighted with the extension for Firefox desktop.  I would also like to be able to use the extension on Firefox mobile.  Are there plans to support the mobile version in the future?

What is the current difficulty? Do Firefox bugs need to be fixed first?

**Comment by Th-o-mas** [comment]
@stefanw 
The questions to Firefox for Android from my previous post are still relevant. Do you have time to give a short answer?

**Comment by stefanw** [comment]
Apparently, there is a FF mobile bug that prevents BibBot from working correctly. There might be a work-around but that would impact all other users (making optional permissions required). Maybe a separate build might work, but that would also need to be distributed separately. I‘m sorry to report that I’m currently not invested in an Android version but will accept PRs.

**Comment by Th-o-mas** [comment]
Thank you Stefan for explaining the issue! Unfortunately I can't contribute code to solve the Firefox for Android bug. However, I (and others) could vote for the importance of the bug at Bugzilla and I could publish a post about it at Mozilla Connect. I would like to do that.

Is there an entry for the Firefox bug on Bugzilla?  Can you give me the link or the number?

**Comment by stefanw** [comment]
Please see this comment: https://github.com/stefanw/bibbot/issues/151#issuecomment-1523169424

**Comment by Th-o-mas** [comment]
@stefanw

Hi Stefan, my post was published on Mozilla Connect:
https://connect.mozilla.org/t5/discussions/better-support-for-the-add-on-for-accessing-journals-and-other/td-p/57062
The post points to the Firefox bug mentioned in #151 that prevents the use of the addon on Firefox for Android.

Anyone who is in favor of fixing the bug and wants to use the addon on Android can: 

- Agree with the post (thumbs up) or
- contribute an answer. 

I hope that this way the bug will get more attention and maybe it will be fixed.

What do you think about linking the post in a prominent place? (https://stefanw.github.io/bibbot/ and https://addons.mozilla.org/en-US/firefox/addon/bibbot/)

**Comment by Th-o-mas** [comment]
@stefanw 

Someone has made a helpful post on Bugzilla pointing to changes already made :-) 
Would you like to take look?
 https://bugzilla.mozilla.org/show_bug.cgi?id=1809917

**Comment by Th-o-mas** [comment]
@stefanw 
Hi Stefan, have you looked at the linked Bugzilla entry for the fixed bug?
Could the addon now work under Firefox for Android?

**Comment by stefanw** [comment]
If you have an Android device, please try it out.

**Comment by guerda** [comment]
I tested out the extension and it partly works.

# Instructions how to install
1. Install Firefox beta (I think) on your Android device
2. Download on your mobile the xpi file for bibbot via this URL: https://addons.mozilla.org/firefox/downloads/file/4284684/bibbot-0.37.0.xpi
3. Go to settings -> about, then tap the firefox logo until debug menu is enabled
4. Go back and select "Install extension from file"
5. Select the xpi you downloaded in 2. 

Source: https://www.androidpolice.com/install-add-on-extension-mozilla-firefox-android/

# Test results
* The addon installs correctly
* It asks for the correct permission
* it opens the genios page if it detects a paywalled article.

The settings menu is completely blank, so I cannot select my correct genios instance and configure the login credentials.
That's why I cannot see if the articles are loaded correctly.

In summary, this is a huge step and I thinks bibbot is on a good way.

**Comment by guerda** [comment]
I noticed that as well (I had about 6 tabs with the settingsvopen and I didn't notice ). I tested it and it works! 

One small thing: if the pay walled article is not the last tab, once bibbot replaces the content, the last tab (not the article) is opened. But that's just a minor nuisance

**Comment by ck2510** [comment]
I was able to use the manually installed .xpi plugin using the same instructions with a standard / non-beta Firefox Mobile version 127.0.2.

**Comment by BenutzerEinsZweiDrei** [comment]
Klappt für mich auch mit der manuellen xpi installation und non beta firefox. Sobald man die bibliothek auswählt fragt er jetzt nach den optional permissions. Dadurch funktionieren Test Seiten.

**Comment by RJdaMoD** [comment]
I tried with 0.40.5 on Firefox 140.0.3 on Android 16:
* The app still identifies as 0.37.0 (maybe because i installed that previously).
* When i try the Zeit sample link, it opens a tab for `https://stadtbib-leipzig.genios.de/searchResult/Alle/Alle%20Quellen?requestText=%22eine+gesellschaftliche+Gruppe+ihre+Anliegen+vorbringt+ihren+Schmerz+ausspricht+wird+davor+gewarnt+dies%22&source=DIE+ZEIT%2CDIE+ZEIT+online%2CZEIT+Campus%2CZEIT+Geschichte%2CZEIT+Wissen%2CChrist+und+Welt`, which results in 403, although login seemed to have worked, and i can manually look up and extract the correct article in the library web interface. I tried to reconstruct the search, this query works: `https://stadtbib-leipzig.genios.de/searchResult/Alle?requestText=%22Auf+dem+R%C3%BCcken+der+Arbeiter+und+der+Armen%22&source=DIE+ZEIT&source=DIE+ZEIT+online&source=ZEIT+Campus&source=ZEIT+Geschichte&source=ZEIT+Wissen&source=Christ+und+Welt`
* The Spiegel sample link does not invoke bibbot, no search tab is opened.
Both links work in desktop browser.

**Comment by neonblind** [comment]
Latest Fennec Update (v 142.0.0) broke bibot for me.

**Comment by neonblind** [comment]
Thanks you.  It seems there are a lot issues with extensions in fennec 142.0.0
https://gitlab.com/relan/fennecbuild/-/issues/134
https://github.com/uBlockOrigin/uBlock-issues/issues/3770

**Comment by neonblind** [comment]
fixed with Fennec 142.0.1
